rs/ethereum/cketh/test_utils/src/ckerc20.rs

```diff
@@ -366,6 +366,21 @@ impl CkErc20Setup {
             .map(|erc20_address| FixedSizeData(erc20_address.into()).to_string())
             .collect()
     }
+
+    pub fn received_erc20_event_topic(&self) -> serde_json::Value {
+        if self.as_ref().evm_rpc_id.is_none() {
+            serde_json::Value::String(RECEIVED_ERC20_EVENT_TOPIC.to_string())
+        } else {
+            // The EVM-RPC canister models topics as `opt vec vec text`, see
+            // https://github.com/internet-computer-protocol/evm-rpc-canister/blob/3cce151d4c1338d83e6741afa354ccf11dff41e8/candid/evm_rpc.did#L69.
+            // This means that a simple topic such as `["0x257e057bb61920d8d0ed2cb7b720ac7f9c513cd1110bc9fa543079154f45f435"]`
+            // must actually be represented as `[["0x257e057bb61920d8d0ed2cb7b720ac7f9c513cd1110bc9fa543079154f45f435"]].
+            // The JSON-RPC providers seem to be able to handle both formats.
```

Comment:
Topics are expressed using an ordered array of data, right?
So, `[A, B]` (A in first position AND B in second position) is not the same as `[[A, B]]` (A or B in first position).

Since the EVM RPC canister expects the format `opt vec vec text`, I think `[A, B]` must be expressed as `[[A], [B]]` (and `[A]` as `[[A]]`).

Reply:
> So, [A, B] (A in first position AND B in second position) is not the same as [[A, B]] (A or B in first position).

That's correct.

> Since the EVM RPC canister expects the format opt vec vec text, I think [A, B] must be expressed as [[A], [B]] (and [A] as [[A]]).

That's correct but I think this is already the case:

1. For ckETH (e.g., `should_not_overlap_when_scrapping_logs`)
    1. Without EVM-RPC canister: `"topics":["0x257e057bb61920d8d0ed2cb7b720ac7f9c513cd1110bc9fa543079154f45f435"]`
    2. With EVM-RPC canister: `"topics":[["0x257e057bb61920d8d0ed2cb7b720ac7f9c513cd1110bc9fa543079154f45f435"]]`
2. For ckERC20 (e.g., `should_scrape_from_last_scraped_after_upgrade`)
    1.  Without EVM-RPC canister: `"topics":["0x4d69d0bd4287b7f66c548f90154dc81bc98f65a1b362775df5ae171a2ccd262b",["0x000000000000000000000000a0b86991c6218b36c1d19d4a2e9eb0ce3606eb48","0x000000000000000000000000dac17f958d2ee523a2206206994597c13d831ec7"]]`
    2. With EVM-RPC canister: `"topics":[["0x4d69d0bd4287b7f66c548f90154dc81bc98f65a1b362775df5ae171a2ccd262b"],["0x000000000000000000000000a0b86991c6218b36c1d19d4a2e9eb0ce3606eb48","0x000000000000000000000000dac17f958d2ee523a2206206994597c13d831ec7"]]`